lib/httparty/request.rb

```diff
@@ -201,7 +201,9 @@ def query_string(uri)
         query_string_parts << options[:query] unless options[:query].nil?
       end
 
-      query_string_parts.reject!(&:empty?) unless query_string_parts == [""]
+      query_string_parts.reject! { |part|
```

Comment:
Hey @jnunemaker thank you for the great review, it was necessary to change only `HTTParty::HashConversions` eventually :)